### ISSUE TITLE
minor new release

### ISSUE THREAD

**Issue by king-11** [dirs-dev/dirs-sys-rs#29]
it has been 8 months since the last release the window-sys package change might be useful further upstream allowing users to fetch only one dependency so if it makes sense can we have a new release?

**Comment by soc** [comment]
Hi @king-11,

> the window-sys package change might be useful further upstream allowing users to fetch only one dependency

which one?

Thanks,

Simon

**Comment by king-11** [comment]
Most of the crates are using windows-sys `0.52` like tokio, request, tempfile, etc. so if `dirs` is used with them then it would pull in `0.48` for now without latest change.

**Comment by ScratchCat458** [comment]
In version `0.59.0` the `HANDLE` type alias has changed: [[1]](https://docs.rs/windows-sys/0.59.0/windows_sys/Win32/Foundation/type.HANDLE.html) [[2]](https://docs.rs/windows-sys/0.52.0/windows_sys/Win32/Foundation/type.HANDLE.html)
```rust
pub type HANDLE = *mut c_void; // 0.59.0 [1]
pub type HANDLE = isize; // 0.52.0 [2]
```
This is backwards incompatible [with this line in `lib.rs`](https://github.com/dirs-dev/dirs-sys-rs/blob/c0fd66cb08f1f97ebf670914253a34bd42d284fb/src/lib.rs#L157) and is a fairly quick fix:
```diff
  let result = Shell::SHGetKnownFolderPath(
              &folder_id,
              0,
-             Win32::Foundation::HANDLE::default(),
+             std::ptr::null_mut(),
              &mut path_ptr
          );
```
Upgrading to support `0.59` would have to come at the cost of being unable to support the current version range (`>= 0.48.0, < 0.53.0`). 

Worth keeping an eye on this regarding what the rest of the ecosystem is using, but unsure how we should proceed.

**Comment by king-11** [comment]
Yeah I think there should be some version that supports a larger set of versions before proceeding onto a bump that would break backwards compatibility.

**Comment by king-11** [comment]
@soc any thoughts how can be proceed to be in sync with the ecosystem referring the latest tokio [Cargo.toml](https://github.com/tokio-rs/tokio/blob/master/tokio/Cargo.toml#L119) the max version is 0.52.0

**Comment by ssrlive** [comment]
How about to publish a new version `dirs-sys 0.5.0` and ship `windows-sys 0.59.0`?

**Comment by king-11** [comment]
It can be a range instead to allow the upstream deps to choose instead amongst two versions if no new feature is getting used.

**Comment by soc** [comment]
Just released released https://crates.io/crates/dirs-sys/0.5.0.

**Comment by king-11** [comment]
@soc are we also getting a `dirs` update as the version of `dirs-sys` it uses is still `0.4.1`

**Comment by soc** [comment]
Yes, I'm on it.

**Comment by soc** [comment]
`dirs` 6.0.0 and `directories` 6.0.0 have been published.
